### ISSUE TITLE
getAttributeNS(xml, 'lang') returns null

### ISSUE THREAD

**Issue by msklvsk** [tmpvar/jsdom#1276]
```xml
<element xml:lang="uk"></element>
```
```javascript
doc.documentElement.getAttributeNS('http://www.w3.org/XML/1998/namespace', 'lang');
// null
```

**Comment by domenic** [comment]
Can you provide a full example of jsdom code I could run by itself?

**Comment by msklvsk** [comment]
Sure,
```javascript
var jsdom = require('jsdom').jsdom;

var doc = jsdom('<element xml:lang="uk"></element>', {
	parsingMode: 'xml'
});

var xmlns = 'http://www.w3.org/XML/1998/namespace';
var lang = doc.documentElement.getAttributeNS(xmlns, 'lang');

console.log(lang);  // null, while 'uk' in browser
```

**Comment by domenic** [comment]
Thanks! There's a hope that this will get fixed in #1247, but I will definitely work on it afterward, in any case. It might be a while though; I am on vacation + at conferences and meetings until November 23.

**Comment by domenic** [comment]
This was not fixed by #1247.

Unfortunately this is an issue with our XML parser, htmlparser2 :(. It only gives its `attribs` object in the form `{ qualifiedName1: value1, qualifiedName2: value2 }`, instead of giving information about namespace and prefix.

That is, it gives `{ 'xml:lang': 'uk' }`, instead of something more useful like `[{ prefix: 'xml', localName: 'lang', namespace: 'http://www.w3.org/XML/1998/namespace', value: 'uk' }]`. I am assuming the XML spec says that the `xml` prefix is automatically mapped to the `http://www.w3.org/XML/1998/namespace` namespace?

So to really fix this, we either need to get htmlparser2 to give more information, or we need to replace our XML parser. Hmm. /cc @fb55 @inikulin.

**Comment by fb55** [comment]
This is out of scope for htmlparser2. https://github.com/isaacs/sax-js handles namespaces, perhaps that's a better alternative for XML parsing.

**Comment by Sebmaster** [comment]
This will be fixed by #1303.
